lib/core/src/lib/datatable/components/date-cell/date-cell.component.ts

```diff
@@ -15,67 +15,53 @@
-                class="adf-datatable-cell-value"
-                *ngIf="format === 'timeAgo'; else standard_date">
-                {{ value$ | async | adfTimeAgo: currentLocale }}
-            </span>
-        </ng-container>
-        <ng-template #standard_date>
-            <span
-                class="adf-datatable-cell-value"
-                title="{{ tooltip | adfLocalizedDate: tooltipDateFormat }}">
-                {{ value$ | async | adfLocalizedDate: format }}
-            </span>
-        </ng-template>
-    `,
+    templateUrl: './date-cell.component.html',
     encapsulation: ViewEncapsulation.None,
-    host: { class: 'adf-date-cell adf-datatable-content-cell' }
+    host: { class: 'adf-date-cell adf-datatable-content-cell' },
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DateCellComponent extends DataTableCellComponent {
+export class DateCellComponent extends DataTableCellComponent implements OnInit {
 
-    static DATE_FORMAT = 'medium';
+    @Input()
+    dateConfig: DateConfig = {};
 
-    currentLocale: string;
-    dateFormat: string;
-    tooltipDateFormat: string;
+    readonly defaultDateConfig: DateConfig = {
+        format: 'medium',
+        tooltipFormat: 'medium',
+        locale: undefined
+    };
 
-    get format(): string {
-        if (this.column) {
-            return this.column.format || this.dateFormat;
-        }
-        return this.dateFormat;
+    constructor(private readonly appConfig: AppConfigService) {
+        super();
     }
 
-    constructor(
-        userPreferenceService: UserPreferencesService,
-        @Optional() dataTableService: DataTableService,
-        appConfig: AppConfigService
-    ) {
-        super(dataTableService);
+    ngOnInit(): void {
+        super.ngOnInit();
+        this.setDefaultConfig();
+    }
 
-        this.dateFormat = appConfig.get('dateValues.defaultDateFormat', DateCellComponent.DATE_FORMAT);
-        this.tooltipDateFormat = appConfig.get('dateValues.defaultTooltipDateFormat', DateCellComponent.DATE_FORMAT);
-        if (userPreferenceService) {
-            userPreferenceService
-                .select(UserPreferenceValues.Locale)
-                .pipe(takeUntil(this.onDestroy$))
-                .subscribe(locale => this.currentLocale = locale);
+    private setDefaultConfig(): void {
+        if (this.dateConfig) {
+            this.dateConfig.format = this.dateConfig?.format ?? (this.column?.format || this.getAppConfigPropertyValue('dateValues.defaultDateFormat', this.defaultDateConfig.format));
+            this.dateConfig.tooltipFormat = this.dateConfig?.tooltipFormat || this.getAppConfigPropertyValue('dateValues.defaultTooltipDateFormat', this.defaultDateConfig.tooltipFormat);
+            this.dateConfig.locale = this.dateConfig?.locale || this.getAppConfigPropertyValue('dateValues.defaultLocale', this.defaultDateConfig.locale);
+        } else {
+            this.dateConfig = this.defaultDateConfig;
```

Comment:
I remember that it was never a good idea to use the same property (input) for storing component state. So add a new property to handle the dateConfig which is different from the input itself

Reply:
replaced by new property new property, thanks